src/hotspot/cpu/aarch64/universalUpcallHandler_aarch64.cpp

```diff
@@ -24,88 +24,314 @@
 
 #include "precompiled.hpp"
 #include "asm/macroAssembler.hpp"
+#include "logging/logStream.hpp"
 #include "memory/resourceArea.hpp"
 #include "prims/universalUpcallHandler.hpp"
+#include "runtime/sharedRuntime.hpp"
+#include "runtime/signature.hpp"
+#include "runtime/signature.hpp"
+#include "runtime/stubRoutines.hpp"
+#include "utilities/formatBuffer.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "vmreg_aarch64.inline.hpp"
 
 #define __ _masm->
 
-// 1. Create buffer according to layout
```

Comment:
The implementation in this file was completely replaced, so it might be easiest to just look at the new code.